Signal/src/ViewControllers/SendExternalFileViewController.m

```diff
@@ -0,0 +1,66 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "SendExternalFileViewController.h"
+#import "Environment.h"
+#import "Signal-Swift.h"
+#import "ThreadUtil.h"
+#import <SignalServiceKit/OWSMessageSender.h>
+#import <SignalServiceKit/TSThread.h>
+
+@interface SendExternalFileViewController () <SelectThreadViewControllerDelegate>
+
+@property (nonatomic, readonly) OWSMessageSender *messageSender;
+
+@end
+
+@implementation SendExternalFileViewController
+
+- (void)loadView
+{
+    [super loadView];
+
+    self.delegate = self;
+
+    _messageSender = [Environment getCurrent].messageSender;
+
+    self.title = NSLocalizedString(@"SEND_EXTERNAL_FILE_VIEW_TITLE", @"Title for the 'send external file' view.");
+}
+
+#pragma mark - SelectThreadViewControllerDelegate
+
+- (void)threadWasSelected:(TSThread *)thread
+{
+    OWSAssert(self.attachment);
+    OWSAssert(thread);
+
+    // We should have a valid filename.
+    OWSAssert(self.attachment.filename.length > 0);
+    NSString *fileExtension = [self.attachment.filename pathExtension].lowercaseString;
+    OWSAssert(fileExtension.length > 0);
+    NSSet<NSString *> *textExtensions = [NSSet setWithArray:@[
```

Comment:
Take it or leave it, but this UX seems a little off to me.

If Alice is "importing" a .txt file, and sends it to Bob, I think Alice expects Bob to see it presented as a txt "file", not as a message bubble, as if she'd "typed" it in.

I think if Alice wanted it presented as a message bubble that she had *typed* in, she would open the txt file in whatever program and copy/paste the contents of it into the text area.